pkg/apis/catalog.cattle.io/v1/types.go

```diff
@@ -6,64 +6,80 @@ import (
 	URL string `json:"url,omitempty"`
 
-	// GitRepo a git repo to clone and index as the helm repo
+	// GitRepo is the git repo to clone which contains the helm repository.
 	GitRepo string `json:"gitRepo,omitempty"`
 
-	// GitBranch The git branch to follow
+	// GitBranch is the git branch where the helm repository is hosted.
 	GitBranch string `json:"gitBranch,omitempty"`
 
 	// CABundle is a PEM encoded CA bundle which will be used to validate the repo's certificate.
 	// If unspecified, system trust roots will be used.
 	CABundle []byte `json:"caBundle,omitempty"`
 
-	// InsecureSkipTLSverify will use insecure HTTPS to download the repo's index.
+	// InsecureSkipTLSverify will use insecure HTTPS to download the helm repository's index file.
 	InsecureSkipTLSverify bool `json:"insecureSkipTLSVerify,omitempty"`
 
-	// ClientSecretName is the client secret to be used to connect to the repo
-	// It is expected the secret be of type "kubernetes.io/basic-auth" or "kubernetes.io/tls" for Helm repos
-	// and "kubernetes.io/basic-auth" or "kubernetes.io/ssh-auth" for git repos.
-	// For a repo the Namespace file will be ignored
+	// ClientSecret is the client secret to be used to connect to the helm repository
+	// It is expected the secret be of type "kubernetes.io/basic-auth" or "kubernetes.io/tls" for Helm repositories
+	// and "kubernetes.io/basic-auth" or "kubernetes.io/ssh-auth" for Github helm repositories.
 	ClientSecret *SecretReference `json:"clientSecret,omitempty"`
 
-	// BasicAuthSecretName is the client secret to be used to connect to the repo
+	// BasicAuthSecretName is the client secret to be used to connect to the helm repository.
 	BasicAuthSecretName string `json:"basicAuthSecretName,omitempty"`
 
-	// ForceUpdate will cause the repo index to be downloaded if it was last download before the specified time
-	// If ForceUpdate is greater than time.Now() it will not trigger an update
+	// ForceUpdate will cause the helm repository index file stored in Rancher
+	// to be updated from the Helm repository URL. This means if there are changes
+	// in the helm repository they will be pulled into Rancher manager.
 	ForceUpdate *metav1.Time `json:"forceUpdate,omitempty"`
 
-	// ServiceAccount this service account will be used to deploy charts instead of the end users credentials
+	// ServiceAccount when specified will be used in creating helm operation pods which in turn
+	// run the helm install or uninstall commands for a chart.
 	ServiceAccount string `json:"serviceAccount,omitempty"`
 
-	// ServiceAccountNamespace namespace of the service account to use. This value is used only on
-	// ClusterRepo and will be ignored on Repo
+	// ServiceAccountNamespace is the namespace of the service account to use.
 	ServiceAccountNamespace string `json:"serviceAccountNamespace,omitempty"`
 
-	// If disabled the repo clone will not be updated or allowed to be installed from
+	// Usage not known yet. Will be removed after review.
```

Comment:
Not sure whether this is a comment to be reviewed pre-merge or we plan to potentially deprecate the field. If the latter, we could keep the previous comment and add an additional DEPRECATED line.

Reply:
@pjbgf That field is not used. I don't even know if it was used before. I think the field was introduced but it was never implemented. I will keep the old comment. I have another story to look into this.